pkg/retriever/cidretrieval.go

```diff
@@ -262,11 +252,6 @@ func (ret *retrieval) runRetrieval(ctx context.Context, candidate RetrievalCandi
 	done := ret.WaitQueue.Wait(queryResponse)
 	defer done()
 
-	if !ret.canSendResult() {
```

Comment:
The point of this being here (and the whole reason for the `canSendResult` channel management weirdness) is so that if we get a query at this point but another goroutine has already performed a full retrieval then we can bail before proceeding. I don't think this implementation is doing that and will proceed to a `retrieveFromCandidate` for every successful query? Or I may be missing something. You have the two selects in line 185 but they're running one directly after the other so are not providing the early-bail.